### PR TITLE
✨ 318 ddd   add a hero to a team

### PR DESCRIPTION
**Pull request by danielmackay**
﻿> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ #318 

> 2. What was changed?

✏️ Added new endpoints to add a hero to a team, and get a specific team

> 3. Did you do pair or mob programming?

✏️ Yes - @matt-goldman-ssw

### CODE REVIEW

File: src/Application/Features/Teams/Commands/AddHeroToTeam/AddHeroToTeamCommand.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public async Task Handle(AddHeroToTeamCommand request, CancellationToken cancellationToken)
+    {
+        var teamId = new TeamId(request.TeamId);
+        var heroId = new HeroId(request.HeroId);
+
+        var team = dbContext.Teams
+            .WithSpecification(new TeamByIdSpec(teamId))
+            .FirstOrDefault();
+
+        if (team is null)
```
Comment (matt-goldman-ssw) [inline src/Application/Features/Teams/Commands/AddHeroToTeam/AddHeroToTeamCommand.cs:23]: I tend to put these checks (team and hero) be in the validator

File: src/Application/Features/Teams/Queries/GetTeam/GetTeamQuery.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public async Task<TeamDto?> Handle(
+        GetTeamQuery request,
+        CancellationToken cancellationToken)
+    {
+        var teamId = new TeamId(request.TeamId);
+
+        var team = await dbContext.Teams
+            .Where(t => t.Id == teamId)
+            .Select(t => new TeamDto
+            {
```
Comment (matt-goldman-ssw) [inline src/Application/Features/Teams/Queries/GetTeam/GetTeamQuery.cs:20]: No Automapper?

File: tests/WebApi.IntegrationTests/Common/Fixtures/IntegrationTestBase.cs [hunk 3]
```diff
@@ -14,6 +14,9 @@ public abstract class IntegrationTestBase : IAsyncLifetime
 
     private readonly TestingDatabaseFixture _fixture;
     protected IMediator Mediator { get; }
+
+    // TODO: Consider removing this as query results can be cached and cause bad test results
+    //       Also, consider encapsulating this and only exposing a `Query` method that internally uses `AsNoTracking()`
```
Comment (matt-goldman-ssw) [inline tests/WebApi.IntegrationTests/Common/Fixtures/IntegrationTestBase.cs:19]:
```suggestion
    //       Also, consider encapsulating this and only exposing a `Query` method that internally uses `AsNoTracking()`
    //       see: https://github.com/SSWConsulting/SSW.CleanArchitecture/issues/324
```

**Review by matt-goldman-ssw — APPROVED** [review]
Looks good.

A couple of comments which you are free to respond to or ignore. One suggestion which I think you should include.

File: src/Application/Features/Teams/Commands/AddHeroToTeam/AddHeroToTeamCommand.cs (hunk 1 above)
Comment (danielmackay) [reply src/Application/Features/Teams/Commands/AddHeroToTeam/AddHeroToTeamCommand.cs:23]: I go back and forward on this.  I used to be of the mind that absolutely all validation should go in the validator.  No I believe that it should only be input validation.  Checking of valid DB IDs, and business logic will happen elsewhere.

File: src/Application/Features/Teams/Queries/GetTeam/GetTeamQuery.cs (hunk 2 above)
Comment (danielmackay) [reply src/Application/Features/Teams/Queries/GetTeam/GetTeamQuery.cs:20]: I probably should have.  Will aim to tidy this up later.

**Pull request merged**